app/controllers/dynamic_pages_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class DynamicPagesController < ApplicationController
+  before_action :require_admin, except: [:index, :show]
+  def index
+    if logged_in?
+      @current_user_is_admin = is_admin?
+      @current_user_is_teacher = is_teacher?
+      @current_user_is_verified_teacher = is_verified_teacher?
+    end
+    @all_dynamic_pages = DynamicPage.all
```

Comment:
In fact, instead of ever calling `all`, we should set a `permissions` variable, and then execute something like `DynamicPage.where(permissions: permissions)` instead.